### DISCUSSION TITLE
Running UE for 5G on RaspberryPi

### DISCUSSION THREAD

**Discussion by tilldroemmer** [srsran/srsRAN_4G#1344] [Q&A]
Hello,

I have been trying for a while now to run the srsRAN_4G UE in 5G mode for the srsRAN_Project setup on a RaspberryPi 4B.
Although I was able to install it, I haven't been able to establish a connection (https://github.com/srsran/srsRAN_Project/issues/442#issue-2112676818).
I have seen the remarks in some issues, that the UE on an RPi is possible in 4G although with low data rates. Did anyone manage to run a 5G UE on an RPi?

Thanks and kind regards,
Till

**Comment by Justin-Garey** [comment]
Hey, I have been able to run the UE on a Raspberry Pi 5 (with 8GB of memory). I was never able to get it to work on the RPi 4.

**Comment by AnilSarode** [comment]
@Justin-Garey May you please share the steps or guide us? Thanks.

**Comment by ohault** [comment]
@Justin-Garey , cool. Can you please share more about your setup using a Raspberry PI 5. I guess @platyman should be interrested too.

**Comment by Justin-Garey** [comment]
Hi @AnilSarode and @ohault, I didn't do anything special. I just followed the [srsRAN guide](https://docs.srsran.com/projects/project/en/latest/tutorials/source/srsUE/source/index.html). When I did it, I used Raspberry Pi OS rather than Ubuntu but at the time, only Ubuntu 23.10 was available for the Pi 5.  Ubuntu 24.04 would likely work now that it seems usable on the Pi 5.

**Reply by Legend-Dd** [reply]
Hi, I managed to set up a 5G SA network on a Raspberry Pi 5, and installed the srsRAN 5G project all on the same Raspberry Pi. When I started the base radio with the b210, I got the following warning:

/home/core5g/srsRAN_Project/lib/pdcp/pdcp_entity_rx.cpp:540: srsran::security::security_result srsran::pdcp_entity_rx::apply_deciphering_and_integrity_check(srsran::byte_buffer, uint32_t):  Assertion failed - Worker index exceeds number of crypto workers. worker_idx=4 max_nof_crypto_workers=4 Aborted

Have you managed to resolve this error?

**Comment by AnilSarode** [comment]
@Justin-Garey Thanks for your response. Which SDR device did you use (any specific name like Lime SDR), I am following [srsRAN 4G](https://docs.srsran.com/projects/4g/en/latest/app_notes/source/pi4/source/index.html) to implement LTE network.

**Reply by Justin-Garey** [reply]
I was using a USRP B200.

**Reply by AnilSarode** [reply]
Thanks @Justin-Garey.
